### PR TITLE
Use hex encoding for plaintext keystore

### LINKED ISSUE #1091: Sort out private key encoding in keystore.json
**Issue summary**

Presently, private keys are encoded in the keystore.json like this:

```
{"default":{"key_type":2,"private_key":[181,166,37,108,245,149,243,31,213,222,199,115,31,216,127,226,255,125,181,179,8,83,223,48,5,69,183,10,238,80,143,95]},"wallet-f3whupgxaxzp2kaboe6pcrzjpaevfxnp7or5igtu2j4q2judqunlicx5ivch3fwgvdhlb3skjzcdq54ccuujsq":{"key_type":2,"private_key":[148,67,3,222,74,51,253,99,250,135,153,224,26,80,242,1,204,77,110,83,248,194,26,76,80,72,170,168,164,252,75,42]},"wallet-f1yfk25vg6jxmpstdj5xlqwkk6hafdibtaed5bxtq":{"key_type":1,"private_key":[25,247,93,49,147,119,31,144,244,233,51,191,44,125,174,185,236,143,65,240,64,125,221,94,192,63,20,58,186,46,88,191]},"wallet-f3qb7njdtw7y4sqoflj74ochlkeewjqbr5kbjxs3rs335kyf7ezo7cistf6uo3eq5qqfiqtawf5efrdsqxhb6a":{"key_type":2,"private_key":[181,166,37,108,245,149,243,31,213,222,199,115,31,216,127,226,255,125,181,179,8,83,223,48,5,69,183,10,238,80,143,95]},"auth-jwt-private":{"key_type":2,"private_key":[176,250,110,195,243,45,45,67,80,123,228,193,34,242,73,66,233,44,245,43,198,121,208,53,227,225,173,147,85,34,54,187]},"wallet-f1ef4cz7bjv67orq5l4s7yfutkfjup7gotn2sqxfa":{"key_type":1,"private_key":[71,12,238,139,237,2,77,48,241,166,57,222,197,141,8,95,156,50,13,51,231,251,178,131,214,122,235,251,155,209,200,207]}}
```

This is aesthetically awful, and I'm not sure how or why it works this way. Earlier I suggested we switch to hex encoding, but apparently lotus encodes keys in their keystore.json via base64.

Complicating matters is that the stdout of `lotus wallet export` is hex-encoded, but I'm not sure what the encoding is in the RPC API. This back-and-forth inconsistency is leading to confusion, more confusion than I would prefer surrounding private keys.

At this point, I think we should match the behavior of lotus, as long as we can make it actually encode properly, and as such, I've changed my mind about hex, and I think we should base64-encode like they do.

**Other information and links**

After further research done by @connormullett on PR #1090, I think the issue is in how we're referencing the string when written. Also, I think we should definitely pretty print the JSON keystore.

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- Use hex encoding to store unencrypted keys instead of a byte vector.

**Reference issue to close (if applicable)**

Closes #1091 

**Other information and links**

### CODE REVIEW

**Comment by cryptoquick** [comment]
There's a lot that's weird here... If base64 was used before, why was it not encoding as such in the keystore.json? Does this code actually change the keystore.json?

File: key_management/Cargo.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ serde_json = "1.0.57"
 serde_cbor = "0.11.1"
 log = "0.4.8"
 sodiumoxide = "0.2.6"
+hex = "0.4.2"
```
Comment (cryptoquick) [inline key_management/Cargo.toml:24]: Have you checked to see if we're using base64 anywhere in this module? If so, you should probably remove it, and also, perhaps look at the features line below

**Comment by cryptoquick** [comment]
After reviewing this more, I think we should stick to base64, if since we were already "trying" to do that.
See the new issue I made to encapsulate this thinking: https://github.com/ChainSafe/forest/issues/1091

**Comment by cryptoquick** [comment]
So, just go ahead and fix the base64 string encoding and we can call it a day, lol.

**Comment by connormullett** [comment]
> So, just go ahead and fix the base64 string encoding and we can call it a day, lol.

Sounds good. To address a previous comment. The `mod json` was only used when exporting the keystore with the `wallet` cli. It wasn't used to write to the persistent keystore.

**Comment by connormullett** [comment]
Moving this PR to #1092.

**Pull request closed without merging**